From 1eddf26dceef76e0a193e450b9e958cd73168f4a Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Mon, 9 Jun 2025 17:21:02 +0800
Subject: [PATCH 207/236] mtk: hostapd: no longer check Milti-Link relation
 when switch the channel of iface

Before single wiphy, STA did the channel switch operation on "phy0" and provide
"radio_idx" so that hostapd can match the target iface and do start
operation on that iface.

Now upstream ucode has added a new input named "radio", which is almost
the same as "radio_idx", expecting that ucode will find the correct iface
to do the operation.

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/ucode.c | 26 +-------------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/src/ap/ucode.c b/src/ap/ucode.c
index 10fcc2b39..40cefcae4 100644
--- a/src/ap/ucode.c
+++ b/src/ap/ucode.c
@@ -660,7 +660,7 @@ uc_hostapd_iface_switch_channel(uc_vm_t *vm, size_t nargs)
 	struct hostapd_config *conf;
 	struct csa_settings csa = {};
 	uint64_t intval;
-	int i, ret = 0, radio_idx;
+	int i, ret = 0;
 
 	wpa_printf(MSG_INFO, "ucode: mtk: channel switch for %s\n", iface->phy);
 	if (!iface || ucv_type(info) != UC_OBJECT)
@@ -696,11 +696,6 @@ uc_hostapd_iface_switch_channel(uc_vm_t *vm, size_t nargs)
 	if ((intval = ucv_int64_get(ucv_object_get(info, "punct_bitmap", NULL))) && !errno)
 		csa.freq_params.punct_bitmap = intval;
 
-	intval = ucv_int64_get(ucv_object_get(info, "radio_idx", NULL));
-	radio_idx = intval;
-	if (errno)
-		radio_idx = iface->current_hw_info ? iface->current_hw_info->hw_idx : 0;
-
 	wpa_printf(MSG_INFO, "ucode: mtk: switch channel information:\n");
 	wpa_printf(MSG_INFO, "    * freq is %d\n", csa.freq_params.freq);
 	wpa_printf(MSG_INFO, "    * bandwidth is %d\n", csa.freq_params.bandwidth);
@@ -712,25 +707,6 @@ uc_hostapd_iface_switch_channel(uc_vm_t *vm, size_t nargs)
 		   csa.freq_params.center_freq2);
 	wpa_printf(MSG_INFO, "    * punct_bitmap is %d\n",
 		   csa.freq_params.punct_bitmap);
-	wpa_printf(MSG_INFO, "    * radio_idx is %d\n", radio_idx);
-
-#ifdef CONFIG_IEEE80211BE
-	for (i = 0; i < iface->interfaces->count; i++) {
-		struct hostapd_iface *tmp_iface = iface->interfaces->iface[i];
-
-		if (tmp_iface->current_hw_info &&
-		    radio_idx == tmp_iface->current_hw_info->hw_idx) {
-			wpa_printf(MSG_INFO,
-				   "ucode: mtk: MLD: switch to iface with radio_idx %d\n",
-				   radio_idx);
-			iface = tmp_iface;
-			break;
-		}
-	}
-
-	if (iface->current_hw_info && radio_idx != iface->current_hw_info->hw_idx)
-		return NULL;
-#endif /* CONFIG_IEEE80211BE */
 
 	for (i = 0; i < iface->num_bss; i++) {
 		ret = hostapd_switch_channel(iface->bss[i], &csa);
-- 
2.45.2

